src/query/service/src/sessions/session_mgr.rs

```diff
@@ -129,34 +123,31 @@ impl SessionManager {
         let session = Session::try_create(id.clone(), typ.clone(), session_ctx, mysql_conn_id)?;
 
         let mut sessions = self.active_sessions.write();
-        if sessions.len() < self.max_sessions {
-            label_counter(
-                METRIC_SESSION_CONNECT_NUMBERS,
-                &config.query.tenant_id,
-                &config.query.cluster_id,
-            );
+        self.validate_max_active_sessions(sessions.len())?;
 
-            if !matches!(typ, SessionType::FlightRPC) {
-                sessions.insert(session.get_id(), Arc::downgrade(&session));
-            }
+        label_counter(
+            METRIC_SESSION_CONNECT_NUMBERS,
+            &config.query.tenant_id,
+            &config.query.cluster_id,
+        );
+        label_gauge(
+            METRIC_SESSION_ACTIVE_CONNECTIONS,
+            sessions.len() as f64,
+            &config.query.tenant_id,
+            &config.query.cluster_id,
+        );
 
-            if let SessionType::MySQL = session_typ {
-                let mut conn_id_session_id = self.mysql_conn_map.write();
-                if conn_id_session_id.len() < self.max_sessions {
-                    conn_id_session_id.insert(mysql_conn_id, id);
-                } else {
-                    return Err(ErrorCode::TooManyUserConnections(
-                        "The current accept connection has exceeded max_active_sessions config",
-                    ));
-                }
-            }
+        if !matches!(typ, SessionType::FlightRPC) {
+            sessions.insert(session.get_id(), Arc::downgrade(&session));
+        }
 
-            Ok(session)
-        } else {
-            Err(ErrorCode::TooManyUserConnections(
-                "The current accept connection has exceeded max_active_sessions config",
-            ))
+        if let SessionType::MySQL = session_typ {
```

Comment:
The  HTTP handle session need check or not?

Reply:
the check logic is still same as before, the modification footprint here is caused by this block of code are all inside a `if else`, this change reduced this `if else`.

Reply:
This question NOT releated to this patch, do you need to check http handler max session?

Reply:
This maybe need to be addressed in another issue, let mark resolved.

Reply:
> This question NOT releated to this patch, do we need to check http handler max session?

I guess the sessions in HTTP / clickhouse can be taken unified in the self.active_sessions check.

but I am curious about why does mysql do the special check? it have both checks about `mysql_conn_map.len()` and `self.active_sessions`

Reply:
Yes, me too. cc @zhang2014